### PR TITLE
fix: simplify OSRelease: stop when identified NON-RHEL

### PR DESCRIPTION
**Pull request by xiangce**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

- Simplify the logic of OSRelease
   Only continue the rest of the identification when cannot identify the OS is NON-RHEL

### CODE REVIEW

File: insights/combiners/os_release.py [hunk 1]
```diff
@@ -100,36 +100,36 @@
...
+    if rhr:
+        if not rhr.is_rhel:
+            return dict(other_linux=rhr.product, reason='NON-RHEL: redhat-release')
+        return dict(other_linux='RHEL')
 
 
 def _from_uname(uname):
     """
-    Internal function to check the `uname -a` output.
+    Identify RHEL by checking the `unam -a`.
```
Comment (JoySnow) [inline insights/combiners/os_release.py:132]: Typo here: `uname -a`

**Review by JoySnow — APPROVED** [review]
Hi @xiangce , the changes looks good to me.

File: insights/combiners/os_release.py [hunk 2]
```diff
@@ -247,54 +249,44 @@ class OSRelease(object):
...
+
+        self._release = 'Unknown'
+        self._reasons = dict(reason='Nothing available to check')
+
+        # Check installed_rpms first
+        # Only when installed_rpms is unavailable, check below specs with order
+        # - uname, dmesg, os_release and redhat_release
+        ret = _from_installed_rpms(rpms, uname)
+
+        if __identify_non_rhel(ret) is None:
```
Comment (huali027) [inline insights/combiners/os_release.py:277]: Here it should continue checking when the `other_linux` is unknown, not only None, right?

Comment (xiangce) [reply insights/combiners/os_release.py:277]: Yup, See L265, this internal function returns `None` only when other_linux is set as `Unknown`, I would add a comment here.

Comment (huali027) [reply insights/combiners/os_release.py:277]: Got it. It looks good to me.

File: insights/combiners/os_release.py (hunk 1 above)
Comment (xiangce) [reply insights/combiners/os_release.py]: Fixed. Thanks.

**Pull request merged**